optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/test/java/org/optaplanner/workbench/screens/domaineditor/backend/server/PlannerDataObjectEditorServiceTest.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.workbench.screens.domaineditor.backend.server;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.workbench.common.services.datamodeller.core.Annotation;
+import org.kie.workbench.common.services.datamodeller.core.DataModel;
+import org.kie.workbench.common.services.datamodeller.core.DataObject;
+import org.kie.workbench.common.services.datamodeller.core.ObjectProperty;
+import org.kie.workbench.common.services.datamodeller.core.impl.AnnotationImpl;
+import org.kie.workbench.common.services.datamodeller.core.impl.DataModelImpl;
+import org.kie.workbench.common.services.datamodeller.core.impl.DataObjectImpl;
+import org.kie.workbench.common.services.datamodeller.core.impl.ObjectPropertyImpl;
+import org.kie.workbench.common.services.datamodeller.util.DriverUtils;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.optaplanner.workbench.screens.domaineditor.model.ComparatorDefinition;
+import org.optaplanner.workbench.screens.domaineditor.model.ComparatorObject;
+import org.optaplanner.workbench.screens.domaineditor.model.ObjectPropertyPath;
+import org.optaplanner.workbench.screens.domaineditor.service.PlannerDataObjectEditorService;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PlannerDataObjectEditorServiceTest {
+
+    private DataModel dataModel;
+
+    private DataObject dataObject1;
+
+    private DataObject dataObject2;
+
+    private PlannerDataObjectEditorService service ;
+
+    @Before
+    public void setUp() {
+        dataModel = getDataModel();
+        service = new PlannerDataObjectEditorServiceImpl();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testInvalidAnnotationFieldPath1() {
```

Comment:
You also don't really need the test prefix anymore if you find your name gets too long otherwise.